Buildkite/ephemeral-agent.md

```diff
@@ -0,0 +1,107 @@
+The agent is created prior to the job execution and deleted once the job completes.
+
+This is achieved by overwriting the Buildkite agent [bootstrap][bootstrap] command of the agent registered.  
+This way we create a wrapper agent that instead of executing the Buildkite job locally, spins up an Orka VM and delegates the job to that VM.
+
+## Requirements
+
+- [Orka][orka] VM
+- Buildkite [agent][agent]
+- [jq][jq] - command-line JSON processor used by the provided scripts
+
+## Set up an Orka VM base image
+
+The Orka VM used as an ephemeral agent needs to have the Buildkite agent installed locally. This is needed to allow the delegation of Buildkite jobs from the wrapper agent.
+
+Since the wrapper agents spins up a new Orka VM, the Buildkite agent must be included in the Orka VM base image.
+
+To include the Buildkite agent in the VM base image:
+1. Set up a new Orka VM. You can Set up an Orka VM using the Orka [CLI][cli] or [REST API][api]. For more information about, see the Orka [quick start guide][quick-start].  
+2. Connect to the Orka VM using VNC.  
+**NOTE**: The VM IP and VNC ports are displayed once the VM is deployed in Orka.  
+3. Verify [Homebrew][homebrew] is installed on the VM. To install Homebrew, see [here][homebrew].
+4. Install the Buildkite agent on the VM by running:  
+    ```
+    brew tap buildkite/buildkite
+    brew install buildkite-agent
+    ```
+5. Save the VM image by running `orka image save` locally.
+
+## Set up an Orka VM config
+
+To allow the wrapper agent to spin up Orka VMs, create an Orka VM config that uses the base image containing the Buildkite agent.  
+A VM config is the template used to spin up a new Orka VM.
+
+You can create up an Orka VM config using the Orka [CLI][cli] or [REST API][api]. For more information about, see the Orka [quick start guide][quick-start].
+
+## Set up a Buildkite agent
+
+To set up a Buildkite wrapper agent:
+1. Install and configure a Buildkite agent on a machine that has network visibility to your Orka environment. For more information on how to install and configure a Buildkite agent, see the [instructions][agent-instructions] provided by Buildkite.
+2. Copy the provided scripts to the agent machine: [base.sh](scripts/base.sh), [bootstrap.sh](scripts/bootstrap.sh), [run.sh](scripts/run.sh). 
+**Note**: All scripts should be in the same directory. For example you can add them to `/usr/local/var/buildkite-agent/`. Overwrite the existing `bootstrap.sh` file if asked.
+3. Verify the scripts can be executed by running `chmod +x path_to_script` in the command line.
+4. Copy the provided [pre-exit hook](scripts/pre-exit) to the agent machine. You should add it to the agent hooks directory. It can be found in the `buildkite-agent.cfg` file under the `hooks-path` property. The location of the `buildkite-agent.cfg` varies depending on your OS. For more information, see your [platform's installation article][agent-instructions].
```

Comment:
to the wrapper agent?

Reply:
The list starts with `To set up the wrapper...` that's it's not explicitly stated. But if you think it could cause confusion - we can add it

Reply:
Same concern as above - people are bound to scan through the document and they might miss out on the context here and there. Better to state wrapper and ephemeral than cause confusion.